### ISSUE TITLE
xtools misbehaves with out of date bootstrap packages.

### ISSUE THREAD

**Issue by knusbaum** [void-linux/void-packages#12907]
### System

* xuname:  
Void 4.19.50_1 x86_64 GenuineIntel notuptodate rmFFFFFFFFFFFFFFFF
* package:  
xbps-src, xtools

### Expected behavior
xgensum and others generate useful messages on error

### Actual behavior
```
$ xgensum -i srcpkgs/cri-tools/template 
/usr/bin/xgensum: line 32: [: too many arguments
/usr/bin/xgensum: line 36: Bootstrap: command not found
/usr/bin/xgensum: line 49: Bootstrap: command not found
```
The behavior comes as a result (i think) of these lines here:
https://github.com/void-linux/void-packages/blob/a47c2dfb5068a9cfb798df67e129b07eba97b22d/xbps-src#L260-L262

Combined with this:
https://git.vuxu.org/xtools/tree/xgensum?id=e697287c4cad3122ba4fd0efdb1a6f5966395324#n16

```
$ xdistdir 
Bootstrap packages must be updated with 'xbps-src bootstrap-update'
```

`xgensum` seems to expect xdistdir to succeed and doesn't check. What's more, the error output from xbps-src goes out on stdout rather than stderr, so the error messages get swallowed.

### Steps to reproduce the behavior
1. Have an out-of-date masterdir in void-packages.
2. try running xgensum.

**Comment by Hoshpak** [comment]
If the error is in xtools and not xbps-src, the issue should be filed with the xtools project, I think https://github.com/leahneukirchen/xtools is the correct upstream repository.

**Comment by Duncaen** [comment]
I think it should be fixed in both, here the error messages should go to stderr and in xtools it should check the exit status.

**Comment by knusbaum** [comment]
Something needs to be done in both I think. I filed the issue here because `xtools`' homepage is listed as being https://git.vuxu.org/xtools where there's no issue tracker, there's a void-packages component to the fix, and because I ran into the bug while using void-packages and thought it likely others would too. I'm happy to file a duplicate ticket against xtools if necessary, if someone confirms the correct place to file.
